megamek/src/megamek/client/ui/swing/util/BASE64ImageView.java

```diff
@@ -51,25 +51,32 @@ private void populateImage() {
             cache = new Hashtable<>();
             getDocument().putProperty("imageCache", cache);
         }
-
         URL src = getImageURL();
-        cache.put(src, loadImage());
-
+        Image image = loadImage();
+        if(image != null) {
+            cache.put(src, image);
+        }
     }
 
     //decodes the Base64 string into an image and returns it
     private Image loadImage() {
         String b64 = getBASE64Image();
-        BufferedImage newImage = null;
-        ByteArrayInputStream bais = null;
-        try {
-            bais = new ByteArrayInputStream(
-                    Base64.getDecoder().decode(b64.getBytes()));
-            newImage = ImageIO.read(bais);
-        } catch (Throwable ex) {
-            ex.printStackTrace();
+        if(b64 != null) {
+            BufferedImage newImage = null;
+            ByteArrayInputStream bais = null;
+            try {
+                bais = new ByteArrayInputStream(
+                        Base64.getDecoder().decode(b64.getBytes()));
+                newImage = ImageIO.read(bais);
+                bais.close();
```

Comment:
Can this be done through try with properties?

Reply:
That's a good call, I'll write it in.